app/handler/provider_complaint.py

```diff
@@ -30,15 +33,72 @@
     Phase,
     ProviderComplaintState,
     RefusedEmail,
+    VerpType,
+    EmailLog,
+    Mailbox,
 )
 
 
+@dataclass
+class OriginalMessageInformation:
+    sender_address: str
+    rcpt_address: str
+    mailbox_address: Optional[str]
+
+
 class ProviderComplaintOrigin(ABC):
     @classmethod
     @abstractmethod
-    def get_original_message(cls, message: Message) -> Optional[Message]:
+    def get_original_addresses(
+        cls, message: Message
+    ) -> Optional[OriginalMessageInformation]:
         pass
 
+    @classmethod
+    def _get_mailbox_id(cls, return_path: Optional[str]) -> Optional[Mailbox]:
+        if not return_path:
+            return None
+        _, return_path = parse_full_address(get_header_unicode(return_path))
+        verp_type, email_log_id = get_verp_info_from_email(return_path)
+        if verp_type == VerpType.transactional:
+            return None
+        email_log = EmailLog.get_by(id=email_log_id)
+        if email_log:
+            return email_log.mailbox.email
+        return None
+
+    @classmethod
+    def sanitize_addresses_and_extract_mailbox_id(
+        cls, rcpt_header: Optional[str], message: Message
+    ) -> Optional[OriginalMessageInformation]:
+        """
+        If the rcpt_header is not None, use it as the valid rcpt address, otherwise try to extract it from the To header
+        of the original message, since in the original message there can be more than one recipients.
+        There can only be one sender so that one can safely be extracted from the message headers.
+        """
+        try:
+            if not rcpt_header:
+                rcpt_header = message[headers.TO]
+            rcpt_list = parse_address_list(get_header_unicode(rcpt_header))
+            if not rcpt_list:
+                saved_file = save_email_for_debugging(message, "NoRecipientComplaint")
+                LOG.w(f"Cannot find rcpt. Saved to {saved_file or 'nowhere'}")
+                return None
+            rcpt_address = rcpt_list[0][1]
+            _, sender_address = parse_full_address(
+                get_header_unicode(message[headers.FROM])
+            )
+
+            return OriginalMessageInformation(
+                sender_address,
+                rcpt_address,
+                cls._get_mailbox_id(message[headers.RETURN_PATH]),
+            )
+        except ValueError:
```

Comment:
Nits: should we move the except closer to where this error can happen? It should only happen in `parse_address_list` right?

Reply:
Yes, but `cls._get_mailbox_id` also calls `parse_address_list`.